compiler/optimizer/LoopReducer.cpp

```diff
@@ -1088,23 +1088,24 @@ TR_LoopReducer::generateArraycopy(TR_InductionVariable * indVar, TR::Block * loo
    if (arraycopyLoop.getStoreAddress()->getIncrement() < 0)
       //&& (storeAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol() == loadAddr->getFirstChild()->getSymbol()->getRegisterMappedSymbol()))
       {
-      if (comp()->cg()->getSupportsPrimitiveArrayCopy())
-         arraycopy->setBackwardArrayCopy(true); /* bit available only to primitive arraycopy */
```

Comment:
I don't think this flag should apply only to primitive array copies. The code just above can definitely create a 5-child arraycopy which is a reference arraycopy. And the original program was copying elements backwards. I don't see what the distinction here between primitive and reference arrays is.

In fact I've checked all the codegens we currently support. No codegen ever checks for the `isBackwardArrayCopy` flag. They all check for `isForwardArrayCopy` and if this returns false we assume its a backwards array copy:

Power:
https://github.com/eclipse/omr/blob/master/compiler/p/codegen/OMRTreeEvaluator.cpp#L3223

x86:
https://github.com/eclipse/omr/blob/master/compiler/x/codegen/OMRTreeEvaluator.cpp#L2894

z/Architecture:
https://github.com/eclipse/omr/blob/master/compiler/z/codegen/OMRTreeEvaluator.cpp#L13481

In addition a few lines below this change we set the `setHalfWordElementArrayCopy` and `setWordElementArrayCopy` flags. Before this change z Systems was the only codegen that answers `true` to `getSupportsPrimitiveArrayCopy` and not surprisingly the above two flags I just mentioned are not at all used withing the z Systems codegen. So there is no reason not to set them for primitive arrays as well.